service-base/src/main/java/org/eclipse/hono/service/plan/NoopResourceLimitChecks.java

```diff
@@ -24,4 +24,10 @@
     public Future<Boolean> isConnectionLimitReached(final TenantObject tenantObject) {
         return Future.succeededFuture(Boolean.FALSE);
     }
+
+    @Override
+    public Future<Boolean> isMessageLimitExceeded(final TenantObject tenantObject,
```

Comment:
Can we rename this to `isDataVolumeLimitReached` and align its semantics with those of *isConnectionLimitReached*?

Reply:
The motive behind keeping _message_ instead of _dataVolume_ as part of the method name is to provide flexibility to the users to choose different attributes to perform the check. For example, the message limit can be based on the number of messages or data volume or combination of both.

Case 1: `bytesAlreadyConsumed + sizeOfIncomingMessage < max-bytes`: returns `false`. The limit is not reached neither exceeded.
Case 2: `bytesAlreadyConsumed + sizeOfIncomingMessage == max-bytes`: still returns `false`. The limit is reached but not exceeded. Thats why I had to name it as `isMessageLimitExceeded` instead of `isMessageLimitReached`.

Reply:
> Case 2: bytesAlreadyConsumed + sizeOfIncomingMessage == max-bytes: still returns false. The limit is reached but not exceeded. Thats why I had to name it as isMessageLimitExceeded instead of isMessageLimitReached.

I don't get it. *isConnectionLimitReached* returns false, if another connection can be established, right? So, analogously, *isMessageLimitReached* should return false if another message can be accepted, right? It is not really important whether `bytesAlreadyConsumed` + `sizeOfIncomingMessage` is less than or equal to the limit as we do not have the exact value of `bytesAlreadyConsumed` for all adapters together anyway. However, the semantics of the two methods would be the same, i.e. if they return false, we accept the current connection attempt/message.

Reply:
I see the point. I will rename it is `isMessageLimitReached`.